include/eos/fitting/ceres_nonlinear.hpp

```diff
@@ -420,31 +445,437 @@ std::array<T, 3> get_shape_point(const morphablemodel::PcaModel& shape_model,
  */
 template <typename T>
 std::array<T, 3> get_vertex_colour(const morphablemodel::PcaModel& color_model, int vertex_id,
-                                   const T* const color_coeffs)
+                                   const T* const color_coeffs, std::size_t num_coeffs_fitting)
 {
-    int num_coeffs_fitting = 10; // Todo: Should be inferred or a function parameter!
     auto mean = color_model.get_mean_at_point(vertex_id);
     auto basis = color_model.get_rescaled_pca_basis_at_point(vertex_id);
     // Computing Colour = mean + basis * coeffs
     // Note: Could use an Eigen matrix with type T to see if it gives a speedup.
-    std::array<T, 3> point{T(mean[0]), T(mean[1]), T(mean[2])};
+    std::array<T, 3> point{static_cast<T>(mean[0]), static_cast<T>(mean[1]), static_cast<T>(mean[2])};
     for (int i = 0; i < num_coeffs_fitting; ++i)
     {
-        point[0] += T(basis.row(0).col(i)(0)) * color_coeffs[i]; // it seems to be ~15% faster when these are
-                                                                 // static_cast<double>() instead of T()?
+        point[0] += static_cast<T>(basis.row(0).col(i)(0)) * color_coeffs[i];
     }
     for (int i = 0; i < num_coeffs_fitting; ++i)
     {
-        point[1] += T(basis.row(1).col(i)(0)) * color_coeffs[i];
+        point[1] += static_cast<T>(basis.row(1).col(i)(0)) * color_coeffs[i];
     }
     for (int i = 0; i < num_coeffs_fitting; ++i)
     {
-        point[2] += T(basis.row(2).col(i)(0)) * color_coeffs[i];
+        point[2] += static_cast<T>(basis.row(2).col(i)(0)) * color_coeffs[i];
     }
     return point;
 };
 
+#endif /* EOS_CERES_USE_OPENCV */
+
+
+ceres::Solver::Options get_default_solver_options() {
+    ceres::Solver::Options solver_options;
+    solver_options.linear_solver_type = ceres::ITERATIVE_SCHUR;
+    solver_options.num_threads = 8;
+    solver_options.num_linear_solver_threads = 8; // only SPARSE_SCHUR can use this
```

Comment:
`num_linear_solver_threads` is actually deprecated in the latest Ceres version and can just be removed.

Reply:
OK. I will remove it.